Me: Drop valueLink

ockham opened the pull request:
A number of files in `/me` use the `form-base` mixin, which has provided a `valueLink` method (i.e. `linkState` drop-in replacement). This PR removes that method in favor of a `getSetting()`, `updateSetting()`, and `toggleSetting()` method, and changes `form-base` consumers to use those methods.

Doesn't use #13925, since these methods are defined in a different file than where they're used. cc @jsnajdr

In addition, this PR removes some other simple `valueLink` cases that aren't suitable for codemodding.

To test: Verify that changing (and saving) settings still works on the following pages:
* http://calypso.localhost:3000/me
* http://calypso.localhost:3000/me/security (Password)
* http://calypso.localhost:3000/me/notifications/subscriptions

Addresses #5951.

jsnajdr commented on client/me/account/main.jsx:
```diff
@@ -60,6 +60,7 @@ const Account = localize(
 	React.createClass( {
 		displayName: 'Account',
 
+		// form-base mixin is needed for getDisabledState() (and possibly other uses?)
```
It would be awesome to replace `form-base` with Redux Form soon ;)

ockham replied:
Indeed, but I didn't want to go down the 🐰 :hole: for the sake of the React 16 upgrade 😄

jsnajdr commented on client/me/form-base/index.js:
```diff
@@ -55,12 +55,19 @@ export default {
+
+	updateSetting: function( settingName ) {
+		return event => {
+			this.props.userSettings.updateSetting( settingName, event.currentTarget.value );
```
You can use the `name` attribute of the target element to determine the field name. All the fields that are affected by the PR already have the right `name`! Then you can attach a single `onChange` handler to everything, instead of creating a new function for every field on every render:
```js
const handleChange = event => {
  const { name, value } = event.target;
  this.props.userSettings.updateSetting( name, value );
}
```
I use this technique in the #13925 codemod.

Of course, your solution still works sufficiently well and if we plan to abandon `form-base` in favor of Redux Form or something similar soon, it's not worth much optimization.

ockham replied:
Addressed in c601a08.

ockham commented:
Thanks for reviewing @jsnajdr! I addressed your feedback; I also added another commit (68b9285) to fix those pesky warnings:

![image](https://user-images.githubusercontent.com/96308/31622025-89e33d14-b29b-11e7-8a73-bec8c6d3c3a9.png)

I think that's the right place to add the fallback value, and since we know where they're used, there shouldn't be any unwanted side effects. LMK if you're okay with this!

jsnajdr commented:
> I think that's the right place to add the fallback value, and since we know where they're used, there shouldn't be any unwanted side effects.

Yes, that looks like the best place for the default. It will also convert a `false` value for a `checked` attribute to empty string. That's not needed, but harmless -- it will be coerced back to boolean when setting a `checked` attribute of an actual DOM element.
